### PR TITLE
Move CRA to app con manager

### PR DESCRIPTION
**Pull request by VOID404**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: application-connector.yaml
Comment (m00g3n) [inline application-connector.yaml:1]: We are missing `app.kubernetes.io/part-of=application-connector-manager` label in `compass-runtime-agent` resources. Without this label the manager will not be able to watch a given resource and will not revert it's properties to original values if they will be changed.

Watch registration code can be found [here](https://github.com/kyma-project/application-connector-manager/blob/main/controllers/applicationconnector_controller.go#L218)

File: controllers/applicationconnector_controller_test.go [hunk 1]
```diff
@@ -83,6 +88,10 @@ func testInstance(t time.Duration, ac v1alpha1.ApplicationConnector) {
 	}
 	Expect(k8sClient.Create(ctx, &gardenerCM)).Should(BeNil())
 
+	By(fmt.Sprintf("create compass-rt-agent configuration: %s/compass-agent-configuration", ac.Namespace))
```
Comment (VOID404) [inline controllers/applicationconnector_controller_test.go:91]: I don't think this is worth shortening, especially in logs...
```suggestion
	By(fmt.Sprintf("create compass-runtime-agent configuration: %s/compass-agent-configuration", ac.Namespace))
```

File: pkg/reconciler/initialize.go [hunk 2]
```diff
@@ -40,5 +40,5 @@ func sFnInitialize(ctx context.Context, r *fsm, s *systemState) (stateFn, *ctrl.
 		return switchState(sFnDeleteResources)
 	}
 
-	return switchState(sFnDetectDomain)
+	return switchState(sFnManageCompassRtAgent)
```
Comment (VOID404) [inline pkg/reconciler/initialize.go:43]: We should remove `nolint:unused` from `sFnManageCompassRtAgent` now

**Review by koala7659 — APPROVED** [review]
LGTM

**Pull request merged**